### ISSUE TITLE
vs. Namecoin?

### ISSUE THREAD

**Issue by natevw** [telehash/blockname#2]
Curious if you have a summary of how this differs from [Namecoin](https://namecoin.info/), either in purpose and/or technique?

**Comment by paulkernfeld** [comment]
In technique, this differs because it uses the existing Bitcoin blockchain, whereas Namecoin has its own blockchain. There is also an argument to be made that Namecoin is unusable because it makes squatting too easy: http://randomwalker.info/publications/namespaces.pdf

I believe that the purpose is approximately the same.

**Comment by xloem** [comment]
It would be great to have a resolver which supports names from both sources!

Namecoin is a little more mature here -- it requires that names expire, which means that squatting incentives decrease when it becomes popular.  Additionally it requires that names be registered in advance via their hash, which protects those registering from having their name stolen by somebody who saw the request but can get a transaction into the blockchain ahead of it.

**Comment by paulkernfeld** [comment]
Yeah... maybe it would be possible to write a DNS resolver that forks out to multiple other DNS resolvers, so that you could plug in whatever domain name registration systems that you like.

@xloem one other comment: since blockname gives the name to the highest bidder, you can't really do a race attack against it. Even if you intercepted someone's registration of a name, you'd still have to pay the same amount in order to take over the name, so being faster wouldn't help you.

**Comment by xloem** [comment]
> @xloem one other comment: since blockname gives the name to the highest bidder, you can't really do a race attack against it. Even if you intercepted someone's registration of a name, you'd still have to pay the same amount in order to take over the name, so being faster wouldn't help you.

That's a significant difference!  I understand that you're saying that the owner of a name is defined by the person who has sent the most coins to it, such that a richer person can always take names from a poorer person in blockname's system.  Hence race attacks are only relevant if you are only willing to pay exactly the same price as somebody else, and squatting is meaningless.

On the other hand, with namecoin, squatting is possible as long as the network continues to allow free renewals, but once you own a name you will know for certain it is reserved for you, and nobody may take it as long as you continue to renew it in the network.  Namecoin is more similar to existing domain name systems, where an earlier participant always has an opportunity to reserve a name from a later participant.

**Comment by paulkernfeld** [comment]
@xloem that is a very good explanation. If anyone is interested in some further discussion of the namespace design issue, the [Kalodner et al. paper](http://randomwalker.info/publications/namespaces.pdf) explores a few different possibilities on page 14.
